### DISCUSSION TITLE
Leaves for TreeLSTM - what are they and what are the options for handling them?

### DISCUSSION THREAD

**Discussion by brando90** [princeton-vl/CoqGym#54] [Q&A]
Hi!

I noticed that leaves were not used in the TreeLSTM. I was curious why that was the case? How are leaves handled and what are the different cases for leaves?

**Comment by yangky11** [comment]
Hi,

Are you talking about the `forward` function here?
https://github.com/princeton-vl/CoqGym/blob/cde1f3f2cd0fa208ea44085e26cae269eec2bc1b/ASTactic/models/term_encoder.py#L137

Nodes are collected by post-order traversing in this line:
https://github.com/princeton-vl/CoqGym/blob/cde1f3f2cd0fa208ea44085e26cae269eec2bc1b/ASTactic/models/term_encoder.py#L145

So they should include leaf nodes?
